**Refactor email service: replace Feign client with WebClient implementation**

Janne6565 opened the pull request:
We are now using a new Mail API which is why i overhauled the MailService Implementation. This was needed as the old api is not available anymore.
Replaced MailClient interface with WebClient for sending emails. Updated email-related configuration properties. Removed no longer used configurations and dependencies. Added WebFlux dependency for the new implementation.

Janne6565 committed:
**Refactor email service: replace Feign client with WebClient implementation**
Replaced MailClient interface with WebClient for sending emails. Updated email-related configuration properties. Removed no longer used configurations and dependencies. Added WebFlux dependency for the new implementation.

EcoFreshKase commented on src/main/resources/application.yaml:
```diff
@@ -14,10 +14,13 @@ cors:
     enabled: true
     admins:
-      - email: "admin@example.org"
+      - email: "jabbekeipert@gmail.com"
```
We can keep it like taht but I would suggest removing your mail address here. The deployment has the credentials for admin creation. Like this everybody would send you mails again. But keep it if you want to no biggie

EcoFreshKase commented on src/main/resources/application.yaml:
```diff
@@ -14,10 +14,13 @@ cors:
     enabled: true
+  mailClient:
+    url: https://jannekeipert.de/mail-service
+    auth-key: ${MAIL_AUTH_KEY:}
```
not sure who this is for local development? Would you provide everybody (if we ever touch this again) with the key?

Also please add it to the readme so one would understand why it is failing

Janne6565 replied:
Yeah, anyone who wants can grab their auth key. Ive added a line to the readme

EcoFreshKase reviewed (changes requested):
lgtm 🥇

Janne6565 committed:
**Update README formatting and improve deployment setup instructions**
Adjusted whitespace and formatting for consistency in the README. Added a step to update the mailClient auth key in the backend setup instructions.

EcoFreshKase reviewed (dismissed):
💯

Janne6565 committed:
**Refactor MailServiceImpl: move WebClient initialization to class level**
Simplified WebClient usage by moving its initialization to a field, reducing redundancy in the email-sending logic.

Janne6565 committed:
**Enhance MailServiceImpl: add error handling for WebClient responses**
Improved email-sending logic by introducing detailed error handling for 4xx and 5xx HTTP responses. Added appropriate logging for success and failure scenarios.
